### PR TITLE
Deployments

### LINKED ISSUE #24: deployment pipeline
🚀

### PR DESCRIPTION
**Pull request by jackyzha0**
closes #24 

changes
* added GA deployment pipeline
* config firebase hosting
* fixed allowed origins on livestream frame

### CODE REVIEW

File: .github/workflows/deploy.yaml
Comment (ianmah) [inline .github/workflows/deploy.yaml:19]: just wondering where the secrets are stored

Comment (jackyzha0) [reply .github/workflows/deploy.yaml:19]: https://github.com/nwplus/livesite/settings/secrets

**Review by ianmah — APPROVED** [review]
LGTM, but I would wait for @Ciuffi to approve before merging

**Review by Ciuffi — APPROVED** [review]
this looks good to me! Does it work properly and everything?

**Comment by Ciuffi** [comment]
Nice, have you tested it?

**Comment by jackyzha0** [comment]
Going to update secrets real quick, but is there a way to test the Github action without merging

**Comment by Ciuffi** [comment]
Not really (besides adding the deployment branch to this action, seeing if it runs, then removing it before merging).

Ideally we should know whether the current firebase settings work properly for a deployment. Deploying it locally should give us this information.

File: .github/workflows/deploy.yaml
Comment (Ciuffi) [inline .github/workflows/deploy.yaml:47]: Not sure how this action works, but do you need the whole firebase string or just the deploy part?

File: src/components/Livestream.js
Comment (Ciuffi) [inline src/components/Livestream.js:58]: Just out of curiosity, Why are you redeclaring the parent query 3 time? Does twitch check each one and take whichever one works, or does it only take the last one?

Comment (jackyzha0) [reply src/components/Livestream.js:58]: Yeah Twitch will just allow all of them as possible origins

File: .github/workflows/deploy.yaml
Comment (jackyzha0) [reply .github/workflows/deploy.yaml:47]: Yeah this action basically just allows you to call the `firebase` CLI from the action so you need the whole string

**Comment by jackyzha0** [comment]
Successfully deployed locally: https://nw-livesite.firebaseapp.com/

File: .github/workflows/deploy.yaml
Comment (Ciuffi) [reply .github/workflows/deploy.yaml:47]: Ah okay, was just a bit confused because the variable there is called args

File: src/components/Livestream.js
Comment (Ciuffi) [reply src/components/Livestream.js]: cool

**Review by Ciuffi — APPROVED** [review]
Looks good to me, good job. Now to see if it works 😉

**Comment by jackyzha0** [comment]
🤞

**Pull request merged**
